src/app/OperationalDeviceProxy.cpp

```diff
@@ -209,7 +218,7 @@ void OperationalDeviceProxy::UpdateDeviceData(const Transport::PeerAddress & add
     }
 }
 
-CHIP_ERROR OperationalDeviceProxy::EstablishConnection()
+CHIP_ERROR OperationalDeviceProxy::EstablishConnection(bool isRecovering)
```

Comment:
It seems like it would be clearer if we just had two separate methods, `EstablishConnection` and `RecoverConnection`.

Reply:
another way is move `MoveToState` call outside.